### PR TITLE
Release v6.0.0-rc1

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -111,3 +111,16 @@ features = ["ansi", "env-filter", "fmt", "time"]
 lto = 'thin'
 panic = 'abort'
 codegen-units = 1
+
+[patch.crates-io]
+bridgetree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }
+incrementalmerkletree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }
```
Comment (nuttycom) [inline Cargo.toml:117]: Since these two crates have releases, we should be able to just depend directly on those released versions.

**Review by nuttycom — DISMISSED** [review]
utACK with possible nit about `incrementalmerkletree` and `bridgetree` versions.

File: doc/book/src/user/release-support.md [hunk 2]
```diff
@@ -42,8 +42,7 @@ time, and may shift due to changes in network solution power.
 <!-- RELEASE_SCRIPT_START_MARKER - If you make changes here, check make-release.py -->
 | `zcashd` version | Release date | Halt height | End of Support |
 | ---------------- | ------------ | ----------- | -------------- |
-| 5.9.1-rc1 | 2024-05-22 | 2643624 | 2024-09-11 |
-| 5.9.1 | 2024-05-23 | 2644900 | 2024-09-12 |
 | 5.10.0-rc1 | 2024-08-22 | 2700600 | 2024-10-31 |
 | 5.10.0 | 2024-08-27 | 2706540 | 2024-11-05 |
+| 6.0.0-rc1 | 2024-09-27 | 2710272 | 2024-11-08 |
```
Comment (daira) [inline doc/book/src/user/release-support.md:47]: This halt height is before the halving (by two weeks), as it should be.

File: Cargo.toml (hunk 1 above)
Comment (str4d) [reply Cargo.toml:117]: I kept this identical to what the revision of `zcash/librustzcash` depends on, to ensure compatibility.

File: src/deprecation.h [hunk 3]
```diff
@@ -14,8 +14,8 @@
 // Shut down nodes running this version of code, `RELEASE_TO_DEPRECATION_WEEKS` weeks' worth
 // of blocks after the estimated release block height. A warning is shown during the 14 days'
 // worth of blocks prior to shut down.
-static const int APPROX_RELEASE_HEIGHT = 2625900;
-static const int RELEASE_TO_DEPRECATION_WEEKS = 10;
+static const int APPROX_RELEASE_HEIGHT = 2661888;
+static const int RELEASE_TO_DEPRECATION_WEEKS = 6;
```
Comment (daira) [inline src/deprecation.h:18]: I checked that this is consistent with `release-support.md`.

**Review by daira — APPROVED** [review]
utACK modulo @nuttycom's comment and the Docker issue.

**Review by nuttycom — APPROVED** [review]
re-utACK c3c5fd5ab1ef5fd4284671b67fd5dd55ccc1c354

**Pull request merged**